src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -177,13 +179,8 @@ protected Future<KafkaClusterStatus> getKafkaConnectionStatus() {
         }
     ).orElseGet(
         () -> {
-            // There is no Kafka cluster configuration, so return the NONE state
-            return Future.succeededFuture(
-                ConnectionStatusKafkaClusterStatusBuilder
-                    .builder()
-                    .state(ConnectedState.NONE)
-                    .build()
-            );
+            // There is no Kafka cluster configuration, so return a null Kafka status
+            return Future.succeededFuture(null);
```

Comment:
And here, too: when there is no `spec.kafka_cluster`, then we want `status.kafka_cluster` to also be null.